### ISSUE TITLE
It's not possible to insert and image in an event's description using an external url

### ISSUE THREAD

**Issue by lecabori** [indico/indico#6255]
Hi all, @OmeGak ,
In the new 3.3 version of Indico, the CKEditorWidget has been replaced with TinyMCEWidget.
I noticed that in this new TinyMCEWidget widget, the constructor by default sets absolute_urls = False.
In an Event's description field it is now not possible to set an absolute path to images attached in description: If I do, it will get converted to an internal url.

**To Reproduce**
1. Go to an Event
2. Click on Settings
3. Edit the section for "Title, Description, Short URL"
4. In the editor, click the "Insert/Edit image" widget icon
5. Enter an absolute path to an image (e.g from the Customisation -> Images menu, get the link from an existing image)
6. Then check the Source code and you'll see that the image src attribute has been changed to an internal path

**Expected behavior**
We would expect that the image source remains as an external url

**Screenshots**
![image](https://github.com/indico/indico/assets/31354117/4145a036-4693-4e33-80c5-e1e9dda495b1)
![image](https://github.com/indico/indico/assets/31354117/dbb88d09-e6f8-40db-a833-7a89137045c7)

**Additional context**
I see that the TinyMCEWidget constructor by default sets absolute_urls = False
See src/indico/web/forms/widgets.py

**Comment by ThiefMaster** [comment]
Are you sure this happens in the *latest* version from Git (not sure which version is installed on your instance)?

I'm asking because in https://github.com/indico/indico/pull/6170 I fixed a problem that converted truly absolute URL to relative ones - it was mainly an issue in emails (where you should never have relative URLs for obvious reasons), but possibly it's the same thing you're encountering here.

```
  relative_urls: false, // kind of misleading: this just ensures domain-relative URLs
  remove_script_host: !forceAbsoluteURLs, // this ensures fully absolute (`https://....`) URLs
```

**Comment by ThiefMaster** [comment]
In any case, if you are on the latest version, and you create an absolute URL that points to the same host where Indico is running: The behavior is intentional. Absolute URLs pointing to the current instance would break if the instance ever changes (sub)domains. So keeping those links relatives makes more sense.

**Comment by lecabori** [comment]
I pulled the latest version today but can't check that the issue has been fixed since I am running into errors.
However I checked the code and the constructor is the same for the TinyMCEWidget
Here is my version and pulls
![image](https://github.com/indico/indico/assets/31354117/14edb2d2-b724-42aa-a006-1522065027bb)

**Comment by lecabori** [comment]
In 3.2.9 it was possible to attach images using external URLs
In our particular case it's important t do so, since we update other sites from Indico and these images won't get displayed

**Comment by ThiefMaster** [comment]
What error are you getting? Could it be related to being on an older Python version (3.3 requires 3.12.2)?

External URLs from **other** hosts than the Indico host should work fine and not be changed.

**Comment by ThiefMaster** [comment]
I just tested on my instance and the current behavior is that:

- URLs pointing to the same host as the Indico instance are always made relative
- URLs pointing to any other host always remain absolute

**Comment by lecabori** [comment]
Well I did upgrade the python version , deleted my existing env and then recreated it to use the newest version.
I then got errors about "No Indico config found. Point the INDICO_CONFIG env var to your config file or move/symlink ...", even though my indico.conf file was still in the right place.
So I simple deleted the whole src folder and reinstalled everything.
All is good, except for the fact that one of my plugins uses a package that has not been updated to use python 3.12.2, so I am stuck there until that is fixed

**Comment by lecabori** [comment]
> I just tested on my instance and the current behavior is that:
> 
> * URLs pointing to the same host as the Indico instance are always made relative
> * URLs pointing to any other host always remain absolute

Yes, but it's different from the behaviour from Indico 3.2.9 where URLs from the same Indico instanced remained as absolute paths if so entered.
In our case we need it to remain absolute so that when we update our other sites with the content from Indico, the images can be correctly rendered

**Comment by ThiefMaster** [comment]
The INDICO_CONFIG error should have been fixed with this commit:

    * 7014cbe8ed - (upstream/master, master) Fix editable package detection (4 hours ago) <Adrian Moennich>

(it happened when installing indico in a new virtualenv without setuptools)

**Comment by lecabori** [comment]
> The INDICO_CONFIG error should have been fixed with this commit:
> 
> ```
> * 7014cbe8ed - (upstream/master, master) Fix editable package detection (4 hours ago) <Adrian Moennich>
> ```
> 
> (it happened when installing indico in a new virtualenv without setuptools)

Hmm, my pull  included that update but I still got the error....

**Comment by ThiefMaster** [comment]
Ugh. Annoying. I guess I'll have to debug this more tomorrow...

**Comment by lecabori** [comment]
Ok, for now I simply deleted src and reinstalled

**Comment by lecabori** [comment]
I see. Thanks, I will take a look
Cheers


* Luis Cabo Rivas*
 Senior Software Developer Perimeter Institute for Theoretical Physics
<http://www.perimeterinstitute.ca>
519-569-7600 x5120
31 Caroline St. N., Waterloo, ON N2L 2Y5
<https://insidetheperimeter.ca/watch-perimeter-equation/>
